metadata-ingestion/src/datahub/sdk/lineage_client.py

```diff
@@ -227,9 +240,140 @@ def add_dataset_transform_lineage(
+            default_db=default_db or "",
+            default_schema=default_schema or "",
+            platform=platform,
+            platform_instance=platform_instance,
+            env=env,
+            graph=self._client._graph,
+            schema_aware=schema_aware,
+        )
+
+        # Validate parsing result
+        if not parsed_result:
+            raise SdkUsageError("Failed to parse SQL query: Unable to parse")
+        if parsed_result.debug_info.error:
+            raise SdkUsageError(
+                f"Failed to parse SQL query: {parsed_result.debug_info.error}"
+            )
+        if not parsed_result.out_tables:
+            raise SdkUsageError(
+                "No output tables found in the query. Cannot establish lineage."
+            )
+
+        # Use the first output table as the downstream
+        downstream_urn = parsed_result.out_tables[0]
+
+        # Process all upstream tables found in the query
+        for i, upstream_table in enumerate(parsed_result.in_tables):
+            # Skip self-lineage
+            if upstream_table == downstream_urn:
+                continue
+
+            # Extract column-level lineage for this specific upstream table
+            column_mapping = {}
+            if parsed_result.column_lineage:
+                for col_lineage in parsed_result.column_lineage:
+                    if not (col_lineage.downstream and col_lineage.downstream.column):
+                        continue
+
+                    # Filter upstreams to only include columns from current upstream table
+                    upstream_cols = [
+                        ref.column
+                        for ref in col_lineage.upstreams
+                        if ref.table == upstream_table and ref.column
+                    ]
+
+                    if upstream_cols:
+                        column_mapping[col_lineage.downstream.column] = upstream_cols
+
+            # Add lineage, including query text only for the first upstream
+            self.add_dataset_transform_lineage(
+                upstream=upstream_table,
+                downstream=downstream_urn,
+                column_lineage=column_mapping or None,
+                query_text=query_text if i == 0 else None,
+            )
+
+    def add_datajob_lineage(
+        self,
+        *,
+        datajob: DatajobUrnOrStr,
+        upstreams: Optional[List[Union[DatasetUrnOrStr, DatajobUrnOrStr]]] = None,
+        downstreams: Optional[List[DatasetUrnOrStr]] = None,
+    ) -> None:
+        """
+        Add lineage between a datajob and datasets/datajobs.
+
+        Args:
+            datajob: The datajob URN to connect lineage with
+            upstreams: List of upstream datasets or datajobs that serve as inputs to the datajob
+            downstreams: List of downstream datasets that are outputs of the datajob
+        """
+
+        datajob_urn = str(datajob)
+        if not datajob_urn.startswith("urn:li:dataJob:"):
```

Comment:
we have helper methods in DataJobUrn for this stuff

Reply:
lmk if i miss anything here -- you mean something like `isinstance(datajob, DataJobUrn)` ? 
isn't that makes this more complicated since we accept `DataJobUrnOrStr` / `DatasetUrnOrStr`?

the workaround I can think of is make everything as general Urn and check instance types -- not sure this is the right approach

Reply:
do something like this https://github.com/datahub-project/datahub/blob/72aab9fe636681c89ce2e69c8dc200e779c5b489/metadata-ingestion/src/datahub/sdk/lineage_client.py#L120

if it's not datajob or whatever it will throw error -- so we can remove all the error handlings